src/app/ui/join-team-page/join-team-page.component.ts

```diff
@@ -13,17 +17,42 @@ export class JoinTeamPageComponent implements OnInit {
 
   teams$: Observable<Team[]>;
 
-  constructor(public auth: AuthService, private teamsService: TeamsService) {}
+  constructor(public auth: AuthService,
+    private teamsService: TeamsService,
+    private notify: NotifyService,
+    private email: EmailService) {}
 
   ngOnInit() {
     this.teams$ = this.teamsService.getTeams();
   }
 
-  applyForTeam(user: User, teamId: string) {
-    this.teamsService.addMembership(user.uid, user.displayName, user.email, teamId);
+  applyForTeam(user: User, teamId: string, teamName: string) {
+    Observable.fromPromise(
+      this.teamsService.addMembership(user.uid, user.displayName, user.email, teamId, teamName)).pipe(
```

Comment:
parameters getting too many here, consider passing the whole user and team objects instead